### PR TITLE
Make the project suitable for pip installation as a package

### PR DESCRIPTION
**Pull request by apmody**
--Figure out directory structure for command line interface
--Figure out directory structure for Web app.

### CODE REVIEW

**Comment by jamesaoverton** [comment]
- Please make sure that everything has been run through black and flake8
- Please rename `tetramer_validator/demo.py` to `tetramer_validator/server.py`
- Did `cli.py` get moved to `tetramer_validator/__main__.py`? I think I'd prefer that it was still called `tetramer_validator/cli.py`.

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    long_description = fh.read()
+setuptools.setup(
+    name="tetramer_validator",
+    version="0.0.1",
+    author="IEDB",
+    author_email="help@iedb.org",
+    description="A small package to validate tetramers",
+    long_description=long_description,
+    long_description_content_type="text/markdown",
+    # url="https://github.com/IEDB/tetramer-validator",
```
Comment (jamesaoverton) [inline setup.py:14]: This should not be a comment. It should either be a proper value (uncommented), or the line should be deleted.

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    classifiers=[
+        "Programming Language :: Python :: 3",
+        # "License :: OSI Approved :: MIT License",
```
Comment (jamesaoverton) [inline setup.py:17]: Does LJI have a policy for this? My default license is BSD3.

File: tetramer_validator/__init__.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import argparse
```
Comment (jamesaoverton) [inline tetramer_validator/__init__.py:1]: I don't like putting shared imports here. Each `.py` file should list its own imports -- it's much more clear.

**Review by jamesaoverton — CHANGES_REQUESTED** [review]
In addition to my three comments on specific line, please also rename `test1.html` to `base.html`. Let me know when these changes are made and I'll test a local install.

**Comment by jamesaoverton** [comment]
Sorry, I thought of one more change that I would like: I think that `molecules.tsv` should only contain the values that you need, which are the first two columns. You can also omit the second and third rows.

File: setup.py (hunk 1 above)
Comment (apmody) [reply setup.py:14]:
```suggestion
    url="https://github.com/IEDB/tetramer-validator",
```

File: setup.py (hunk 2 above)
Comment (jamesaoverton) [reply setup.py:17]: I guess we could not have a license for now.

Comment (apmody) [reply setup.py:17]:
```suggestion
```

File: tetramer_validator/__init__.py (hunk 3 above)
Comment (apmody) [reply tetramer_validator/__init__.py]: Changed import statements to match this.  https://github.com/IEDB/tetramer-validator/pull/17/commits/aed27624f28c69115926be90267d39ae2bf00ddf

**Comment by jamesaoverton** [comment]
I forgot that you are filtering `molecules.tsv` for partial and complete molecules. I think it should be filtered before you save it, and then you don't need the Restriction column.

File: tetramer_validator/data/molecule.tsv [hunk 4]
```diff
@@ -0,0 +1,695 @@
+Label	IEDB Label	Synonyms	Restriction Level
```
Comment (jamesaoverton) [inline tetramer_validator/data/molecule.tsv:1]: I don't think that the Restriction Level column is needed. Please remove it.

**Review by jamesaoverton — CHANGES_REQUESTED** [review]
Ok, I can `pip install -e .` this and run `tv ...`. I'm still having some problems:

1. `pytest` does not see `tests/tests.py`. It should be renamed `tests/test_validate.py`. When I rename it and run `pytest`, all the tests fail.
2. When I run `flake8 --max-line-length 100 --ignore E203,W503 tetramer_validator/*` (as recommended [here](https://github.com/knocean/practises/tree/master/python#style)) I get many errors.

File: tetramer_validator/data/molecule.tsv (hunk 4 above)
Comment (apmody) [reply tetramer_validator/data/molecule.tsv]: https://github.com/IEDB/tetramer-validator/pull/17/commits/d4c8f2b1cb2d2bb7c78b64648c5c778e1426c84f

**Review by jamesaoverton — APPROVED** [review]
I fixed the Flake8 warnings and made sure that all the files use Unix line endings (some were using DOS line endings).

This is good to go. Please merge it when you're ready.

**Pull request merged**